### DISCUSSION TITLE
how to get all routes by request

### DISCUSSION THREAD

**Discussion by pincman** [fastify/fastify#4288] [Q&A] [answered]
I using fastify in nestjs,and in controller mtehod ,I can get fastify request,If I want to get all routes from request,how to do it?

```typescript
   @Get('test')
    async test(@Request() req: FastifyRequest) {
      // how to get all routes here
        return request.xxx.getRoutes
    }
```

**Comment by Uzlopak** [comment]
You want to enumerate all routes which are registered in the router?

**Comment by mcollina** [comment, marked as answer]
Thanks for reporting. We do not have the resources to support Nest.js users. Please refer to the Nest [Discord](discord.gg/NestJS) channel (support) for such questions.
